### DISCUSSION TITLE
Representing topological relationships of meshed surfaces

### DISCUSSION THREAD

**Discussion by pshriwise** [mfem/mfem#3990] [Q&A]
I'm interested in translating some capability for particle transport on CAD-based surface mesh to MFEM. It relies on topological relathionships between surfaces to form volumes. The layout of the model looks like this:

<img width="1003" alt="image" src="https://github.com/mfem/mfem/assets/4563941/3a8723ea-08d3-4d05-bf66-a7e1de387099">

The mesh is divided into groups of triangles and each group represents a surface. The volume is defined by relationships between its group and the surfaces the compose the volume. The volume groups are empty -- no triangles contained there.

This is a lean organization for the model b/c shared surfaces only need to be represented once and we can move particles from one volume to another based on these topological relathionships in the mesh model.

Is there a way to represent this type of structure in MFEM?

**Comment by tzanio** [comment]
ping: @jandrej

**Reply by jandrej** [reply]
Please see the [SubMesh examples](https://mfem.org/examples/#multidomain-and-submesh-demonstration-miniapp). You should be able to represent your diagram above with that. I don't fully understand the volume definition. Can you show us an example mesh? We're happy to help if there are any more specific questions.

**Reply by pshriwise** [reply]
Thanks @jandrej! In this example, are the elements at the boundary of the two domains shared by the domains?

**Reply by jandrej** [reply]
They are duplicated and the sharing part is up to the user implementation, e.g. "how do the different physics communicate with each other".

**Comment by v-dobrev** [comment]
If the volume boxes do not actually contain any volume elements, then one can represent this as a surface mesh in MFEM where the different surface parts are marked with different attributes.

If the volume boxes contain volume elements, then this can be represented as one connected mesh where volume elements are assigned attributes based on the volume they belong to; the surfaces (including surfaces between different volumes) will be described as boundary elements where each boundary element is assigned an attribute based on the surface it belongs to.

Given such (connected) mesh, one can then extract sub-components based on the attributes using the `SubMesh` class.

**Reply by pshriwise** [reply]
Excellent! I might try a proof-of-concept example for representing our geom topology soon. Thanks @v-dobrev.
